zerver/lib/actions.py

```diff
@@ -6579,16 +6590,16 @@ def get_active_presence_idle_user_ids(
     if realm.presence_disabled:
         return []
 
-    is_pm = message_type == "private"
+    is_private_message = message_type == "private"
 
     user_ids = set()
-    for user_id in active_user_ids:
-        flags = user_flags.get(user_id, [])
-        mentioned = "mentioned" in flags or "wildcard_mentioned" in flags
-        private_message = is_pm and user_id != sender_id
-        alerted = "has_alert_word" in flags
-        if mentioned or private_message or alerted:
-            user_ids.add(user_id)
+    for user_data in active_users_data:
+        alerted = "has_alert_word" in user_data.flags
+
+        # Assume all users are idle for notifiability. We will later filter out those
+        # users who aren't idle.
```

Comment:
I think this could express the idea much more clearly like so:
"""
We only need to know the presence idle state for a user if this message would be notifiable
for them if they were indeed idle. Only including those users in the calculation below is a
very important optimization for open communities with many inactive users.
"""

Reply:
Done